File: solutions/java/word-count/1/src/main/java/WordCount.java
import java.util.HashMap;
import java.util.Map;

public class WordCount {

    Map<String, Integer> wordCount;
    String punct = "!!&@$%^&";
    WordCount() {
        wordCount = new HashMap();
    }

    Map<String, Integer> phrase (String sentence) {
        sentence = sentence.replaceAll("( )+", " ");
        sentence = sentence.replaceAll("(!|&|@|\\$|%|\\^|&|\\.)+", "");
        sentence = sentence.trim();
        sentence = sentence.toLowerCase();
        System.out.println("sentence="+sentence);
        for (String word : sentence.split("(\n| |,|:)+", 0)) {
            System.out.println("word="+word);
            System.out.println(word.substring(0,1)+word.substring(word.length()-1,word.length()));
            if (word.substring(0,1).equals("'") && word.substring(word.length()-1,word.length()).equals("'")){
                word = word.substring(1,word.length()-1);
                //System.out.println("word="+word);
            }
            if (wordCount.get(word) == null) {
                wordCount.put(word, 1);
            } else {
                wordCount.put(word, wordCount.get(word)+1);
            }
        }

        return(wordCount);
    }
}

File: solutions/javascript/word-count/1/word-count.js
//
// This is only a SKELETON file for the 'Word Count' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const countWords = (phrase) => {
  var wc={};
  var nphrase = phrase.toLowerCase().replace(/[:!&@$%^&.]+/g,'').replace(/  /g," ").trim();
  var words = nphrase.split(/[ \n\t,]+/);
  for (var wi =0; wi < words.length; wi++) {
    words[wi]=words[wi].replace(/^["']*|["']*$/g,'')
    if (words[wi] != "") {
      if (words[wi] in wc) {
        wc[words[wi]]++;
      } else {
        wc[words[wi]] = 1;
      }
    }
  }
  return wc;
};
